src/test/java/org/dcsa/api_validator/bkg/v1/shipments/testGetShipmentByCarrierBookingReference.java

```diff
@@ -0,0 +1,93 @@
+
+import static org.dcsa.api_validator.bkg.v1.BookingTestConfiguration.JSON_SCHEMA_VALIDATOR;
+import static org.hamcrest.Matchers.*;
+
+/*
+ * Tests related to the GET /shipments/{carrierBookingReference} endpoint
+ */
+
+public class testGetShipmentByCarrierBookingReference {
+
+    private List<String> carrierBookingReferenceListForShipments;
+
+    @BeforeClass
+    public void getAllCarrierBookingReferencesForShipments() {
+        this.carrierBookingReferenceListForShipments =
+                JsonPath.from(
+                                given()
+                                        .auth()
+                                        .oauth2(Configuration.accessToken)
+                                        .get(Configuration.ROOT_URI + SHIPMENT_SUMMARIES_PATH)
+                                        .then()
+                                        .assertThat()
+                                        .statusCode(200)
+                                        .body("size()", greaterThanOrEqualTo(0))
+                                        .body(JSON_SCHEMA_VALIDATOR)
+                                        .extract()
+                                        .body()
+                                        .asString())
+                        .getList("carrierBookingReference");
+    }
+
+    @Test
+    public void testGetValidShipments() {
+        carrierBookingReferenceListForShipments.forEach(
+                carrierBookingReference ->
+                        given()
+                                .auth()
+                                .oauth2(Configuration.accessToken)
+                                .when()
+                                .pathParam("carrierBookingReference", carrierBookingReference)
+                                .get(Configuration.ROOT_URI + SHIPMENT_CARRIERBOOKINGREFERENCE_PATH)
+                                .then()
+                                .assertThat()
+                                .body("size()", greaterThanOrEqualTo(0))
+                                .body("carrierBookingReference", equalTo(carrierBookingReference))
+                                .header("API-Version", "1.0.0")
+                                .statusCode(HttpStatus.SC_OK)
+                                .body(JSON_SCHEMA_VALIDATOR));
+                }
+
+    @Test
+    public void testShipmentWithInvalidCarrierBookingReference() {
+        given()
+                .auth()
+                .oauth2(Configuration.accessToken)
+                .when()
+                .pathParam(
+                        "carrierBookingReference",
+                        "12345678912345678901235678945651205451686156465154515564845156754845678465544567845648456548151554234")
+                .get(Configuration.ROOT_URI + SHIPMENT_CARRIERBOOKINGREFERENCE_PATH)
+                .then()
+                .assertThat()
+                .statusCode(HttpStatus.SC_BAD_REQUEST);
```

Comment:
I've bumped up the version in DCSA-BKG to include the expected error format, maybe we can add the expected body in the test as well?

Reply:
Update: We will discuss later on how we want to go about it. We can leave it as is for now,